cli/tester.js

```diff
@@ -0,0 +1,275 @@
+  }
+  return temporalUnit;
+}
+
+const readRecipientAddress = () => {
+  let toAddress = readlineSync.question(`Enter the recipient address: [press enter for ${web3.eth.defaultAccount}]\n`)
+  if (!toAddress) {
+    toAddress = web3.eth.defaultAccount
+  }
+
+  // Validate the address
+  toAddress = ethUtil.addHexPrefix(toAddress)
+  if (!eac.Util.checkValidAddress(toAddress)) {
+    console.log("Not a valid address")
+    console.log("[FATAL] exiting...")
+    process.exit(1)
+  }
+
+  return toAddress
+}
+
+const readWindowStart = currentBlockNumber => {
+  const defaultWindowStart = currentBlockNumber + defaultSchedulingValues.minimumPeriodBeforeSchedule + 5
+  const windowStart = readlineSync.question(`Enter window start: [press enter for ${defaultWindowStart}]\n`)
+
+  return windowStart || defaultWindowStart
+}
+
+const pick = (obj, keys) => {
+  return keys.map(k => k in obj ? {[k]: obj[k]} : {})
+             .reduce((res, o) => Object.assign(res, o), {});
+}
+
+const main = async (_) => {
+  if (program.stats) {
+    let rawTx = []
+    try {
+      rawTx = fs.readFileSync('scheduled.txt','utf8')
+    } catch (e) {
+      console.log("Unable to read scheduled.txt file")
+      process.exit(1)
+    }
+
+    const executedAfter = (executedAt, tx) => {
+      const windowStart = tx.windowStart * 1
+      executedAt = executedAt * 1
+
+      return executedAt - windowStart
+    }
+
+    const tx = rawTx.split('\n')
+    const requests = await Promise.all(tx.filter(t => !!t).map(async (t) => {
+      const tx = eac.transactionRequest(t)
+      await tx.fillData()
+
+      const res = pick(tx, ["address", "windowStart", "claimedBy", "requiredDeposit", "bounty", "wasSuccessful"])
+      const { blockNumber } = await tx.getExecutedEvent()
+      const accuracy = ((blockNumber * 1) - (tx.windowStart * 1)) / (tx.wind)
+
+      return Object.assign(res, {executedAfter: executedAfter(blockNumber, tx)})
+    }))
+
+    console.table(requests)
+  } else {
+    let scheduleParams = await getDefaultSchedulingValues()
+    if (!await eac.Util.checkNetworkID()) {
+      console.log("  error: must be running a localnode on the Ropsten or Kovan networks")
+      process.exit(1)
+    }
+    if (!await eac.Util.checkForUnlockedAccount()) process.exit(1)
+    if (program.json) scheduleParams = JSON.parse(program.json)
+
+    const eacScheduler = await eac.scheduler()
+
+    // Starts the scheduling wizard.
+    clear()
+    console.log("🧙 🧙 🧙  Schedule a transaction  🧙 🧙 🧙\n")
+
+    const temporalUnit = scheduleParams.temporalUnit || readTemporalUnit()
+    const toAddress = scheduleParams.recipient || web3.eth.defaultAccount
+    const callData = scheduleParams.callData || "0x0"
+    const callGas = scheduleParams.callGas
+    const callValue = scheduleParams.callValue
+
+    const currentBlockNumber = await eac.Util.getBlockNumber()
+
+    const windowStart = scheduleParams.windowStart || currentBlockNumber + scheduleParams.minimumPeriodBeforeSchedule + 5
```

Comment:
Can you put parenthesis around `currentBlockNumber + scheduleParams.minimumPeriodBeforeSchedule + 5`?

I've already found one crucial bug in libs, because of using `||` and math operators without parenthesis. It gets very tricky.